File: CS/SecuredDataServer/ServerApplication.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp.MiddleTier;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Xpo;
using XafModule;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.Xpo.Metadata;

namespace DataServer {
    public class ExampleServerApplication : ServerApplication {
        public ExampleServerApplication() {
            ApplicationName = "ServerUsageExample";
            this.Modules.Add(new ExampleModule());
        }
        protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
            args.ObjectSpaceProvider = new XPObjectSpaceProvider(args.ConnectionString, args.Connection);
        }
    }
}

File: VB/SecuredDataServer/ServerApplication.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.ExpressApp.MiddleTier
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports XafModule
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports DevExpress.Xpo.Metadata

Namespace DataServer
	Public Class ExampleServerApplication
		Inherits ServerApplication
		Public Sub New()
			ApplicationName = "ServerUsageExample"
			Me.Modules.Add(New ExampleModule())
		End Sub
		Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
			args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection)
		End Sub
	End Class
End Namespace
